packages/editor-ui/src/stores/__tests__/assistant.store.test.ts

```diff
@@ -0,0 +1,320 @@
+			type: 'message',
+			role: 'assistant',
+			text: 'Hello!',
+		};
+		assistantStore.addAssistantMessages([message], '1');
+		expect(assistantStore.chatMessages.length).toBe(1);
+		expect(assistantStore.chatMessages[0]).toEqual({
+			id: '1',
+			type: 'text',
+			role: 'assistant',
+			content: 'Hello!',
+			quickReplies: undefined,
+			read: false,
+		});
+	});
+
+	it('can add an assistant message with quick replies', () => {
+		const assistantStore = useAssistantStore();
+
+		const message: ChatRequest.MessageResponse = {
+			type: 'message',
+			role: 'assistant',
+			text: 'Hello!',
+			quickReplies: [
+				{ text: 'Yes', type: 'text' },
+				{ text: 'No', type: 'text' },
+			],
+		};
+		assistantStore.addAssistantMessages([message], '1');
+		expect(assistantStore.chatMessages.length).toBe(1);
+		expect(assistantStore.chatMessages[0]).toEqual({
+			id: '1',
+			type: 'text',
+			role: 'assistant',
+			content: 'Hello!',
+			read: false,
+			quickReplies: [
+				{ text: 'Yes', type: 'text' },
+				{ text: 'No', type: 'text' },
+			],
+		});
+	});
+
+	it('can add an assistant code-diff message', () => {
+		const assistantStore = useAssistantStore();
+
+		const message: ChatRequest.MessageResponse = {
+			type: 'code-diff',
+			role: 'assistant',
+			description: 'Here is the suggested code change',
+			codeDiff: 'diff --git a/file1 b/file2',
+			suggestionId: '1',
+			solution_count: 1,
+			quickReplies: [
+				{ text: 'Yes', type: 'text' },
+				{ text: 'No', type: 'text' },
+			],
+		};
+		assistantStore.addAssistantMessages([message], '1');
+		expect(assistantStore.chatMessages.length).toBe(1);
+		expect(assistantStore.chatMessages[0]).toEqual({
+			id: '1',
+			type: 'code-diff',
+			role: 'assistant',
+			description: 'Here is the suggested code change',
+			codeDiff: 'diff --git a/file1 b/file2',
+			suggestionId: '1',
+			read: false,
+			quickReplies: [
+				{ text: 'Yes', type: 'text' },
+				{ text: 'No', type: 'text' },
+			],
+		});
+	});
+
+	it('can add an assistant summary message', () => {
+		const assistantStore = useAssistantStore();
+
+		const message: ChatRequest.MessageResponse = {
+			type: 'summary',
+			role: 'assistant',
+			title: 'Summary',
+			content: 'Here is the summary',
+			quickReplies: [
+				{ text: 'Yes', type: 'text' },
+				{ text: 'No', type: 'text' },
+			],
+		};
+		assistantStore.addAssistantMessages([message], '1');
+		expect(assistantStore.chatMessages.length).toBe(1);
+		expect(assistantStore.chatMessages[0]).toEqual({
+			id: '1',
+			type: 'block',
+			role: 'assistant',
+			title: 'Summary',
+			content: 'Here is the summary',
+			read: false,
+			quickReplies: [
+				{ text: 'Yes', type: 'text' },
+				{ text: 'No', type: 'text' },
+			],
+		});
+	});
+
+	it('can add an agent suggestion message', () => {
+		const assistantStore = useAssistantStore();
+
+		const message: ChatRequest.MessageResponse = {
+			type: 'agent-suggestion',
+			role: 'assistant',
```

Comment:
Same as above? Maybe we can have a create assistant message and provide as parameter the type, which will create the object with default values and accept parameters in case you want to override values for a specific Uds case 🤔

Reply:
Same as above :)